core/src/main/java/kafka/server/ReplicaFetcherTierStateMachine.java

```diff
@@ -68,14 +68,15 @@
  */
 public class ReplicaFetcherTierStateMachine implements TierStateMachine {
     private static final Logger log = LoggerFactory.getLogger(ReplicaFetcherTierStateMachine.class);
-
+    private  BrokerTopicStats brokerTopicStats;
     private LeaderEndPoint leader;
     private ReplicaManager replicaMgr;
 
     public ReplicaFetcherTierStateMachine(LeaderEndPoint leader,
                                           ReplicaManager replicaMgr) {
         this.leader = leader;
         this.replicaMgr = replicaMgr;
+        this.brokerTopicStats = replicaMgr.brokerTopicStats();
```

Comment:
This would currently break a lot of tests because `MockTierStateMachine` passes `null` for the `replicaMgr` parameter of the `ReplicaFetcherTierStateMachine` constructor.

Reply:
Good catch. Thanks